### PR TITLE
#29 Make DB engines as optional dependencies

### PR DESCRIPTION
**Pull request by Kostanos**

### CODE REVIEW

**Comment by mcostanzo** [comment]
👏 👏 👏

File: package.json [hunk 1]
```diff
@@ -19,10 +19,7 @@
   },
   "dependencies": {
     "async": "0.2.x",
-    "mssql": "2.x.x",
-    "mysql": "2.x.x",
-    "newline": "0.0.3",
-    "pg": "^4.5.5"
```
[the remark this answers is not in the document]
Comment (Kostanos) [reply package.json]: I do not checking the version, but only if dependency is installed inside lib/create-common-client.js

I would say that we should keep maintaining this lib with latest versions of those dependencies, if someone has issue, he/she can report it and we update.

[the remark this answers is not in the document]
Comment (rickbergfalk) [reply package.json]: I don't know if there's a reliable way to do this without each target package exposing the version. Sequelize requires the driver to be installed separately and seems to be working for them... not sure how it handles changes in major driver versions though.

**Comment by rickbergfalk** [comment]
I'm going to publish a postgrator patch release to npm prior to merging this in. 

This will be considered a breaking change so I'm thinking it might be worth going through and taking this opportunity to make other breaking changes and/or cleanup.

**Comment by rickbergfalk** [comment]
Thanks @Kostanos - Merging this in but will hold off publishing. Probably will be next week some time at earliest

**Pull request merged**